lib/ansible/playbook/conditional.py

```diff
@@ -133,7 +133,7 @@ def _check_conditional(self, conditional, templar, all_vars):
             disable_lookups = hasattr(conditional, '__UNSAFE__')
             conditional = templar.template(conditional, disable_lookups=disable_lookups)
             if bare_vars_warning and not isinstance(conditional, bool):
-                display.deprecated('evaluating %r as a bare variable, this behaviour will go away and you might need to add |bool'
+                display.deprecated('evaluating %r as a bare variable, this behaviour will go away and you might need to add " is truthy"'
```

Comment:
It would probably be good to mention this filter, but it doesn't always work.
For example:
```yaml
---
- hosts: localhost
  gather_facts: no
  vars:
    str_value: 'false'
    nested:
      str_value: 'false'
  tasks:
  - command: echo should not see
    when: str_value

  - command: echo should not see
    when: nested.str_value
```
The fact that the second task executes can be resolved by using | bool to keep the same behavior for the first task, and fix the bug with the second:
```yaml
---
- hosts: localhost
  gather_facts: no
  vars:
    str_value: 'false'
    nested:
      str_value: 'false'
  tasks:
  - command: echo should not see
    when: str_value | bool

  - command: echo should not see
    when: nested.str_value | bool
```
If the `is truthy` filter is used instead, both of those tasks would be executed.

Reply:
@bcoca pointed out that is_truthy does take a `convert_bool` argument, so that would fix that ^

Reply:
Oh, man.  I thought the evaluation of bare variables in `when` was the same way Python does with variables in `if` or `bool()`.  Did it use to be so?  And is it exactly the same as `| bool` at the moment?  Then I would have to change the message again.

Reply:
it was 'jinja' evaluation, which was mostly pythonic, but Ansible also follows 'yaml booleans' which makes it more complicated .. specially when the `bool` filter is doing an incomplete version of that .. see #70721